apps/studio/src/server/modules/resource/resource.router.ts

```diff
@@ -474,54 +497,44 @@ export const resourceRouter = router({
       return query.select(["role"]).execute()
     }),
 
-  getAncestryOf: protectedProcedure
-    .input(getAncestrySchema)
+  getAncestryWithSelf: protectedProcedure
+    .input(getAncestryWithSelfSchema)
     .query(async ({ input: { siteId, resourceId } }) => {
       if (!resourceId) {
         return []
       }
+      return await getAncestryWithSelf({
+        siteId: Number(siteId),
+        resourceId: Number(resourceId),
+      })
+    }),
 
-      const ancestors = await db
-        .withRecursive("Resources", (eb) =>
-          eb
-            .selectFrom("Resource")
-            .select([
-              "Resource.id",
-              "Resource.title",
-              "Resource.permalink",
-              "Resource.parentId",
-            ])
-            .where("Resource.siteId", "=", Number(siteId))
-            .where("Resource.id", "=", resourceId)
-            .unionAll(
-              eb
-                .selectFrom("Resource")
-                .innerJoin("Resources", "Resources.parentId", "Resource.id")
-                .select([
-                  "Resource.id",
-                  "Resource.title",
-                  "Resource.permalink",
-                  "Resource.parentId",
-                ]),
-            ),
-        )
-        .selectFrom("Resources")
-        .select([
-          "Resources.id",
-          "Resources.title",
-          "Resources.permalink",
-          "Resources.parentId",
-        ])
-        .execute()
-
-      return ancestors.reverse().slice(0, -1)
+  getBatchAncestryWithSelf: protectedProcedure
+    .input(getBatchAncestryWithSelfSchema)
+    .query(async ({ input: { siteId, resourceIds } }) => {
+      const ancestryPromises = resourceIds.map(async (id) => {
+        return await getAncestryWithSelf({
+          siteId: Number(siteId),
+          resourceId: Number(id),
+        })
+      })
+      return await Promise.all(ancestryPromises)
     }),
 
   search: protectedProcedure
     .input(searchSchema)
     .output(searchOutputSchema)
     .query(
-      async ({ ctx, input: { siteId, query = "", cursor: offset, limit } }) => {
+      async ({
+        ctx,
+        input: {
+          siteId,
+          query = "",
+          resourceTypes = Object.values(ResourceType),
```

Comment:
suggest not setting defaults on the params itself, but inside the schema for a single source of truth. you can use zod schema's [`.default`](https://zod.dev/?id=default) function.